### ISSUE TITLE
Test failure due to 360 degrees not being recognized as having the same meaning as zero degrees.

### ISSUE THREAD

**Issue by wafels** [sunpy/sunpy#2919]
### Description

A bug showed up in the CI for #2916.

________________________________ test_hcrs_hgs _________________________________
```
    def test_hcrs_hgs():
        # Get the current Earth location in HCRS
        now = Time(parse_time('now'))
        earth_hcrs = SkyCoord(get_body_barycentric('earth', now), frame='icrs', obstime=now).hcrs
    
        # Convert from HCRS to HGS
        earth_hgs = earth_hcrs.transform_to(HeliographicStonyhurst)
    
        # The HGS longitude of the Earth should be zero within numerical error
>       assert quantity_allclose(earth_hgs.lon, 0*u.deg, atol=1e-12*u.deg)
E       assert False
E        +  where False = quantity_allclose(<Longitude 360. deg>, (0 * Unit("deg")), atol=(1e-12 * Unit("deg")))
E        +    where <Longitude 360. deg> = <SkyCoord (HeliographicStonyhurst: obstime=2019-01-31 15:27:09.378755): (lon, lat, radius) in (deg, deg, AU)\n    (360., -5.9683332, 0.98522121)>.lon
E        +    and   Unit("deg") = u.deg
E        +    and   Unit("deg") = u.deg

sunpy/coordinates/tests/test_transformations.py:99: AssertionError
```

### Expected behavior

360 degrees is also zero degrees.  Probably this test should not have failed

### Actual behavior

It did fail!

### System Details

==========================================================
 SunPy Installation Information

==========================================================

###########
 General
###########
Time : Thursday, 31. January 2019 04:02PM UT
System : Darwin
Processor : i386
Arch : 64bit
SunPy : 0.9.2
SunPy_git : eded238635695f1c28a8b8e29bf1aa7de0bcee05
OS: Mac OS X 10.13.6 (i386)

###########
 Required Libraries 
###########
Python: 3.6.8
NumPy: 1.15.4
SciPy: 1.1.0
matplotlib: 3.0.2
Astropy: 3.1.1
Pandas: 0.24.0

###########
 Recommended Libraries 
###########
beautifulsoup: 4.7.1
PyQt: NOT INSTALLED
SUDS: 0.6
Sqlalchemy: 1.2.16
Requests: 2.21.0

**Comment by nabobalis** [comment]
I think I got a hack fix in #2911.

**Comment by nabobalis** [comment]
Well I only fix the test, you want to fix the actual problem that's someone else's issue.
